pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +271,84 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+
+	partitionList, err := t.GetPartitionsFromTimeRange(txn, startTime, endTime)
+	if err != nil {
+		return resources, stats, errors.Wrapf(err, "failed to get partitions from table:%v, from startTime:%v, to endTime:%v", t.tableName, startTime, endTime)
+	}
+
+	tablePrefix := "/" + t.tableName + "/"
+	iterOpt := badger.DefaultIteratorOptions
+	itr := txn.NewIterator(iterOpt)
+	defer itr.Close()
+
+	lastPartition := ""
+	for _, currentPartition := range partitionList {
+		curPartitionPrefix := tablePrefix + currentPartition + "/"
+		itr.Seek([]byte(curPartitionPrefix))
```

Comment:
We want to seek to the keyPrefix if it has at least /table/partition/, else just use /table/partition/